### PR TITLE
Links to buttons

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue # .

### Summary of Changes
This replaces link tags that look and act like buttons with actual buttons. See #20682 for details.

### Testing Instructions
Code review.

### CODE REVIEW

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 6401c60d86e4bbead2d167e7c94fb5886f25340f<br /><br />Code Review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22494">issues.joomla.org/tracker/joomla-cms/22494</a>.</sub>

File: administrator/components/com_templates/views/template/tmpl/default_modal_folder_footer.php [hunk 1]
```diff
@@ -13,7 +13,7 @@
 ?>
 <form id="deleteFolder" method="post" action="<?php echo JRoute::_('index.php?option=com_templates&task=template.deleteFolder&id=' . $input->getInt('id') . '&file=' . $this->file); ?>">
 	<fieldset>
-		<a href="#" class="btn" data-dismiss="modal"><?php echo JText::_('COM_TEMPLATES_TEMPLATE_CLOSE'); ?></a>
+		<button type="button" class="btn" data-dismiss="modal"><?php echo JText::_('COM_TEMPLATES_TEMPLATE_CLOSE'); ?></button>
 		<input type="hidden" class="address" name="address" />
 		<?php echo JHtml::_('form.token'); ?>
 		<input type="submit" value="<?php echo JText::_('COM_TEMPLATES_BUTTON_DELETE'); ?>" class="btn btn-danger" />
```
Comment (Quy) [inline administrator/components/com_templates/views/template/tmpl/default_modal_folder_footer.php:19]: Convert to button?

File: administrator/components/com_contact/views/contacts/tmpl/default_batch.php [hunk 2]
```diff
@@ -52,10 +54,10 @@
 		</div>
 	</div>
 	<div class="modal-footer">
-		<button class="btn" type="button" onclick="document.getElementById('batch-category-id').value='';document.getElementById('batch-access').value='';document.getElementById('batch-language-id').value='';document.getElementById('batch-user-id').value='';document.getElementById('batch-tag-id').value=''" data-dismiss="modal">
+		<button type="button" class="btn" onclick="document.getElementById('batch-category-id').value='';document.getElementById('batch-access').value='';document.getElementById('batch-language-id').value='';document.getElementById('batch-user-id').value='';document.getElementById('batch-tag-id').value=''" data-dismiss="modal">
 			<?php echo JText::_('JCANCEL'); ?>
 		</button>
-		<button class="btn btn-primary" type="submit" onclick="Joomla.submitbutton('contact.batch');">
+		<button type="submit" class="btn btn-primary" onclick="Joomla.submitbutton('contact.batch');">
```
Comment (Quy) [inline administrator/components/com_contact/views/contacts/tmpl/default_batch.php]: In other instances, this button is `btn-success` except here.

File: administrator/components/com_templates/views/template/tmpl/default_modal_folder_footer.php (hunk 1 above)
Comment (SharkyKZ) [reply administrator/components/com_templates/views/template/tmpl/default_modal_folder_footer.php:19]: Maybe some other time.

File: administrator/components/com_contact/views/contacts/tmpl/default_batch.php (hunk 2 above)
Comment (SharkyKZ) [reply administrator/components/com_contact/views/contacts/tmpl/default_batch.php]: This one I'll revert. This file is not used and should have been deleted after #7003 was implemented. Can it be just deleted now or is there some special procedure for this?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on e65cff574874987ff041652dc7d51476b73541d2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22494">issues.joomla.org/tracker/joomla-cms/22494</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22494">issues.joomla.org/tracker/joomla-cms/22494</a>.</sub>

**Comment by brianteeman** [comment]
woohoo

**Comment by brianteeman** [comment]
@SharkyKZ would you like to do a similar pr for J4?

**Comment by SharkyKZ** [comment]
I thought everything that goes into 3.x is supposed to reach 4.0 eventually. I can see a lot of a11y stuff going into 4.0 now. But, IMO, simple markup fixes like this should be done against 3.x too.

**Comment by brianteeman** [comment]
you are correct but we have a really big backlog on that and I would really like to get as much a11y stuff done in J4 as quickly as possible. Plus I suspect that a lot of the remaining links->buttons are modal related and we plan to change from a bootstrap modal at some point so it would be good to have them changed now as it will be hard to add later from the 3.x branch

**Comment by SharkyKZ** [comment]
Should this be done for both 3.x and 4.0 then? Or just 4.0?

**Pull request merged**